beacon-chain/core/helpers/rewards_penalties.go

```diff
@@ -69,7 +69,21 @@ func IncreaseBalance(state *stateTrie.BeaconState, idx uint64, delta uint64) err
 	if err != nil {
 		return err
 	}
-	return state.UpdateBalancesAtIndex(idx, balAtIdx+delta)
+	return state.UpdateBalancesAtIndex(idx, IncreaseBalanceWithVal(balAtIdx, delta))
+}
+
+// IncreaseBalanceWithVal increases validator with the given 'index' balance by 'delta' in Gwei.
+// This method is flattened version of the spec method, taking in the raw balance and returning
+// the post balance.
+//
+// Spec pseudocode definition:
+//  def increase_balance(state: BeaconState, index: ValidatorIndex, delta: Gwei) -> None:
+//    """
+//    Increase the validator balance at index ``index`` by ``delta``.
+//    """
+//    state.balances[index] += delta
+func IncreaseBalanceWithVal(currBalance uint64, delta uint64) uint64 {
+	return currBalance + delta
```

Comment:
While it might seem excessive to have a separate function, I think having a separate method for this is fine as it follows `DecreaseBalanceWithVal` and also corresponds to the spec helper. We should add more comments to the godoc explaining why it is different from the spec method.